InpsydeTemplates/Sniffs/Formatting/ShortEchoTagSniff.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace InpsydeTemplates\Sniffs\Formatting;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * @psalm-type Token = array{
+ *     type: string,
+ *     code: string|int,
+ *     line: int
+ *     }
+ */
+final class ShortEchoTagSniff implements Sniff
+{
+    /**
+     * @return list<int|string>
+     */
+    public function register(): array
+    {
+        return [
+            T_ECHO,
+        ];
+    }
+
+    /**
+     * @param File $phpcsFile
+     * @param int $stackPtr
+     *
+     * phpcs:disable Inpsyde.CodeQuality.ArgumentTypeDeclaration
+     */
+    public function process(File $phpcsFile, $stackPtr): void
+    {
+        // phpcs:enable Inpsyde.CodeQuality.ArgumentTypeDeclaration
+
+        /** @var array<int, Token> $tokens */
+        $tokens = $phpcsFile->getTokens();
+        $currentLine = $tokens[$stackPtr]['line'];
+
+        $prevPtr = $phpcsFile->findPrevious(
+            Tokens::$emptyTokens,
+            ($stackPtr - 1),
+            null,
+            true
+        );
+
+        if (!is_int($prevPtr) || !isset($tokens[$prevPtr])) {
+            return;
+        }
+
+        $prevToken = $tokens[$prevPtr];
+
+        if ($prevToken['line'] !== $currentLine) {
+            return;
+        }
+
+        if ($prevToken['code'] !== T_OPEN_TAG) {
+            return;
+        }
```

Comment:
We only need `$currentLine` after the next `if`, and we may want to switch the two `if` statements:

```suggestion

        $prevPtr = $phpcsFile->findPrevious(
            Tokens::$emptyTokens,
            ($stackPtr - 1),
            null,
            true
        );

        if (!is_int($prevPtr) || !isset($tokens[$prevPtr])) {
            return;
        }

        $prevToken = $tokens[$prevPtr];

        if ($prevToken['code'] !== T_OPEN_TAG) {
            return;
        }

        $currentLine = $tokens[$stackPtr]['line'];

        if ($prevToken['line'] !== $currentLine) {
            return;
        }
```